.github/workflows/nightly-tests.yml

```diff
@@ -43,22 +43,25 @@ jobs:
           args: '--all-targets'
 
   test_release:
-    runs-on: ubuntu-22.04
+    runs-on: warp-ubuntu-2404-x64-4x
     needs: check_if_needs_running
     if: needs.check_if_needs_running.outputs.status > 0
 
     steps:
     - uses: actions/checkout@v4
       with:
         submodules: recursive
-    - name: ⚡ Cache rust
-      uses: actions/cache@v4
+    - name: ⚡ Restore rust cache
+      id: cache
+      uses: WarpBuilds/cache/restore@v1
       with:
         path: |
-          ~/.cargo/registry
-          ~/.cargo/git
-          target
-        key: ${{ runner.os }}-cargo-release-${{ hashFiles('**/Cargo.toml') }}
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+          Cargo.lock
+        key: ${{ runner.os }}-cargo-nightly-tests
```

Comment:
What is the reasoning for this change?

Reply:
using the same cache as pr-tests, Lucas changed it from time to time so I assume this is more optimized